### PR TITLE
Refactor stack-monitor tests

### LINKED ISSUE #141: Can we make the tests USN order-independent?
In https://github.com/paketo-buildpacks/stack-usns/pull/136, as well as https://github.com/paketo-buildpacks/stack-usns/commit/9bb4df2584a80b80429e08689bfd9222297d9085 we have had to change the order of `affected_packages` in our [test assertions](https://github.com/paketo-buildpacks/stack-usns/blob/main/actions/usn-monitor/entrypoint/main_test.go) whenever the USN RSS feed we monitor changes the order in which they publish dependencies. 

It would be nice if we didn't have tests that had to be changed every time a third-party changes the order of items we are consuming. We should investigate if we can refactor these tests somehow to work regardless of the order of USNs and their internal fields such as `affected_packages`, and then implement these changes to cut down on tech debt associated with these tests.

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Resolves #141. The USN monitoring tests used to fail once in a while because they rely on the output of HTML on an external Ubuntu USN page to be in a certain order. This PR refactors the stack USNS monitoring tests.

The old tests used a mock server to mock out what we get from https://ubuntu.com/security/notices/rss.xml, but then used real USNs in that server, and actually reached out to the real links for each USN to test that we could retrieve CVEs and other information. This refactor completely mocks out all of the HTML for the entire process, so that our tests are only testing the HTML parsing functionality in the code, and not dependent on the content of a website we don't maintain.

I also added a `query_rss_test.go` which actually does reach out to https://ubuntu.com/security/notices/rss.xml, and then just makes sure that we get some kind of output XML that we can generate USN entries for. We don't care what the content is, just that we can ping the server and successfully perform a transformation on the output information to get it into an expected format. This test is slow because it actually reads and retrieve CPEs/other information for each USN, and depending on the USNs present that day, there could be a large number of CPEs. I believe we need this test to make sure that the RSS feed is still reachable and serves output as expected.

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [ ] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by sophiewigmore** [comment]
Tagged @ryanmoran for additional review to @paketo-buildpacks/stacks-maintainers, as he provided some insights into why we should refactor.

**Comment by brayanhenao** [comment]
It looks good for me, lets wait until @ryanmoran reviews it.

**Review by brayanhenao — APPROVED** [review]

**Review by ryanmoran — APPROVED** [review]

**Pull request merged**
